pgjdbc/src/main/java/org/postgresql/util/StreamWrapperFinalizeAction.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2023, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.util;
+
+import static org.postgresql.util.internal.Nullness.castNonNull;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * The action deletes temporary file in case the user submits a large input stream,
+ * and then abandons the statement.
+ */
+class StreamWrapperFinalizeAction implements Closeable {
+  private @Nullable InputStream stream;
+  private @Nullable Path tempFile;
+
+  StreamWrapperFinalizeAction(Path tempFile) {
+    this.tempFile = tempFile;
+  }
+
+  public InputStream getStream() throws IOException {
+    InputStream stream = this.stream;
+    if (stream == null) {
+      stream = Files.newInputStream(castNonNull(tempFile));
```

Comment:
will a call to `getStream` after a call to `close` result in a NullPointerException?
is that expected/desired?

Reply:
StreamWrapper is an internal one-time thing. They should not call `getStream` after `close`